File: test-suite/src/test/java/io/micronaut/picocli/docs/config/ConfigDemoTest.java
package io.micronaut.picocli.docs.config;

import io.micronaut.configuration.picocli.PicocliRunner;
import io.micronaut.context.ApplicationContext;
import io.micronaut.context.env.Environment;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ConfigDemoTest {

    @Test
    void testConfigurationExample() {

        try (ApplicationContext ctx = ApplicationContext.run(Environment.CLI, Environment.TEST)) {
            String[] args = new String[] { };
            Object called = PicocliRunner.call(ConfigDemo.class, ctx, args);

            assertEquals("Hi!", called);
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/picocli/docs/config/ConfigDemoTest.kt
package io.micronaut.picocli.docs.config

import io.micronaut.configuration.picocli.PicocliRunner
import io.micronaut.context.ApplicationContext
import io.micronaut.context.env.Environment
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ConfigDemoTest {

    @Test
    fun testConfigurationExample() {
        ApplicationContext.run(Environment.CLI, Environment.TEST).use { ctx ->
            val args = arrayOf<String>()
            val called = PicocliRunner.call(ConfigDemo::class.java, ctx, *args)

            assertEquals("Hi!", called)
        }
    }
}
